src/v/cluster/scheduling/allocation_node.h

```diff
@@ -89,20 +93,22 @@ class allocation_node {
     }
     bool is_full() const { return _allocated_partitions >= _max_capacity; }
     allocation_capacity max_capacity() const { return _max_capacity; }
-    ss::shard_id allocate();
+    ss::shard_id allocate(bool prio);
 
 private:
     friend allocation_state;
 
-    void deallocate(ss::shard_id core);
-    void allocate(ss::shard_id core);
+    void deallocate_on(ss::shard_id core, bool prio);
```

Comment:
`bool` args are often frowned upon in favor of `ss::bool_class` or other strong type, though I'll leave it to others to comment if this is an absolute rule or not.

Reply:
Great point! The `bool` flag has been a kind of a draft in here, as I wanted to roll out the PR ASAP (while the issue has had a kind of urgency attached to it) to have its logic reviewed in the first place.